elab-util-extra/src/Language/Reflection/Unify/WithCompiler.idr

```diff
@@ -367,11 +367,15 @@ unifyWithCompiler task = do
 export
 unifyWithCompiler' :
   Elaboration m =>
-  MonadError UnificationError m =>
+  MonadError (Maybe UnificationError) m =>
   UnificationTask ->
   m $ UnificationResult
 unifyWithCompiler' task = do
-  rr <- runEitherT {m} {e=UnificationError} $ unify' task
-  dg <- liftEither rr
+  dg <- unify' task
   pure $ finalizeDG task dg
 
+export %defaulthint
+UnifyWithCompiler : Elaboration m => Unify m
```

Comment:
This named implementation shouldn't have `%defaulthint`, `%defaulthint` should be added to `Unify.idr`, something like
```idris
%defaulthint %inline
DefaultUnifier : ... => Unify m
DefaultUnifier = UnifyWithCompiler
```